### PR TITLE
Replace BazelPackageInfo usage with new Interface during import

### PR DESCRIPTION
**Pull request by simontoens**
This is to allow different import code paths - the one based on projectview
file changes won't construct BazelPackageInfo instances.

### CODE REVIEW

**Comment by simontoens** [comment]
This is for #5

File: plugin-core/src/main/java/com/salesforce/bazel/eclipse/config/BazelEclipseProjectFactory.java [hunk 1]
```diff
@@ -196,20 +196,12 @@ private static void importBazelWorkspacePackagesAsProjects(BazelPackageInfo pack
...
             importedProjectsList.add(eclipseProject);
         }
-
-        for (BazelPackageInfo childPackageInfo : packageInfo.getChildPackageInfos()) {
-            try {
-                importBazelWorkspacePackagesAsProjects(childPackageInfo, bazelWorkspaceRoot, importedProjectsList);
-            } catch (Exception anyE) {
-                anyE.printStackTrace();
-            }
-        }
```
Comment (simontoens) [inline plugin-core/src/main/java/com/salesforce/bazel/eclipse/config/BazelEclipseProjectFactory.java]: I removed this because I don't think we use it, and it simplifies the import logic: the list of bazel packages to import is always a flat list of, conceptually, fully qualified paths to bazel packages.  This is how the UI works today anyway.  And when importing because of projectview file changes, that will also be the case.

File: plugin-libs/plugin-model/src/main/java/com/salesforce/bazel/eclipse/model/BazelPackageLocation.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import java.io.File;
+
+/**
+ * Minimal representation of a Bazel Package location on the file system.
+ * 
+ * @see BazelLabel for a container that is also Bazel Target aware. 
+ * @author stoens
+ * @since March 2020
+ */
+public interface BazelPackageLocation {
```
Comment (simontoens) [inline plugin-libs/plugin-model/src/main/java/com/salesforce/bazel/eclipse/model/BazelPackageLocation.java]: I copied the important methods, needed during import, from BazelPackageInfo into this interface.  I considered changing the method names a little, but left them alone for now to make it more obvious what is going on.  Rename these can be a different PR.

File: plugin-core/src/main/java/com/salesforce/bazel/eclipse/config/BazelEclipseProjectFactory.java (hunk 1 above)
Comment (plaird) [reply plugin-core/src/main/java/com/salesforce/bazel/eclipse/config/BazelEclipseProjectFactory.java]: Cool. On the topic of the flat list, at some point we will want to order it. We want the first projects to be imported to be the upstream deps of the latter imported projects. Not part of project views, but that is on the horizon. That will be the key to speeding up perf of import.
